src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/EcmaFormat/ReflectionTypeProvider.cs

```diff
@@ -0,0 +1,288 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Runtime.InteropServices;
+using System.Reflection.Metadata;
+using System.Reflection.Runtime.General;
+using System.Reflection.Runtime.TypeInfos;
+
+namespace System.Reflection.Runtime
+{
+    static class ReflectionTypeProviderHelpers
```

Comment:
Do we need a separate class for this? Can it go into MetadataReaderExtensions.Ecma.cs?